contracts/governance/Staking/Staking.sol

```diff
@@ -216,14 +233,34 @@ contract Staking is
      * @param stakeFor The address to stake the tokens for or 0x0 if staking for oneself.
      * @param delegatee The address of the delegatee or 0x0 if there is none.
      * */
-    function stakesBySchedule(
+    function stakeBySchedule(
         uint256 amount,
         uint256 cliff,
         uint256 duration,
         uint256 intervalLength,
         address stakeFor,
         address delegatee
-    ) public whenNotPaused {
+    ) external whenNotPaused {
+        _stakeBySchedule(amount, cliff, duration, intervalLength, stakeFor, delegatee);
+    }
+
+    /**
+     * @notice Stake tokens according to the vesting schedule.
+     * @param amount The amount of tokens to stake.
+     * @param cliff The time interval to the first withdraw.
+     * @param duration The staking duration.
+     * @param intervalLength The length of each staking interval when cliff passed.
+     * @param stakeFor The address to stake the tokens for or 0x0 if staking for oneself.
+     * @param delegatee The address of the delegatee or 0x0 if there is none.
+     * */
+    function _stakeBySchedule(
```

Comment:
pls ignore changes in Staking and all its inheritance contracts
the changes were forced by some functions changes so that not to run into compiler errors to keep Staking and inheritance in the codebase so far to facilitate comparison when reviewing
the refactoered `stakeBySchedule` is in the `StakingStakeModule` contracts

Reply:
thanks, reviewed it and it looks good